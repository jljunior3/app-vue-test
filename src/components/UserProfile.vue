<template>
  <div class="user-card">
    <span class="user-name">{{ user.name }}</span>
    <span class="user-email">{{ user.email }}</span>
  </div>
</template>

<script>
export default {
  name: 'UserProfile',
  props: {
    user: {
      type: Object,
      required: true
    }
  },
  created() {}
}
</script>
<style lang="scss">
.user-card {
  display: flex;
  flex-direction: column;
  margin: 10px;
  width: 300px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
  transition: 0.3s;
  background-color: #fff;

  &:hover {
    box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
  }

  .user-name {
    font-size: 1.2em;
    margin: 5px;
  }

  .user-email {
    font-size: 0.9em;
    margin-bottom: 5px;
  }
}
</style>
