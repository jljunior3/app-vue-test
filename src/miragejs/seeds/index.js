const usersSeeder = server => {
  server.createList('user', 10);
};

export default function seeds(server) {
  usersSeeder(server);
}
