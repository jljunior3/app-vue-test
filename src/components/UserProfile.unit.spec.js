import { mount } from '@vue/test-utils'
import { makeServer } from '@/miragejs/server'
import UserProfile from './UserProfile.vue'

describe('UserProfile - unit', () => {
  let server

  beforeEach(() => {
    server = makeServer({ environment: 'test' })
  })

  afterEach(() => {
    server.shutdown()
  })

  it('should match snapshot', () => {
    const wrapper = mount(UserProfile, {
      propsData: {
        user: server.create('user', {
          name: 'Júnior',
          email: 'teste@teste.com.br'
        })
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  it('should mount the component', async () => {
    const wrapper = mount(UserProfile, {
      propsData: {
        user: server.create('user', {
          name: 'Júnior',
          email: 'teste@teste.com.br'
        })
      }
    })
    expect(wrapper.vm).toBeDefined()
    expect(wrapper.text()).toContain('Júnior')
    expect(wrapper.text()).toContain('teste@teste.com.br')
  })

  it('should trigger a commit when the button is clicked', async () => {
    const mockStore = { commit: jest.fn() }
    const user = {
      id: 1,
      name: 'Júnior',
      email: 'teste@teste.com.br'
    }
    const wrapper = mount(UserProfile, {
      propsData: {
        user
      },
      mocks: {
        $store: mockStore
      }
    })

    await wrapper.find('[data-testid="button-card"]').trigger('click')

    expect(mockStore.commit).toHaveBeenCalledWith('SET_USER_SELECTED', user)
  })
})
