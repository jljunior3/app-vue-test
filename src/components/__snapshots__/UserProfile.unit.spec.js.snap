// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile - unit should match snapshot 1`] = `
<div
  class="user-card"
>
  <span
    class="user-name"
  >
    JÃºnior
  </span>
   
  <span
    class="user-email"
  >
    teste@teste.com.br
  </span>
</div>
`;
